java/ql/test/library-tests/dataflow/taintsources/FileUpload.java

```diff
@@ -0,0 +1,46 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.FileItemStream;
+
+
+public class FileUpload {
+
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+    private javax.servlet.http.Part filePart;
+    private FileItem fileItem;
+    private FileItemStream fileItemStream;
+    private jakarta.servlet.http.Part jakartaPart;
+
+    private static void sink(Object o) {}
+
+    public void test() throws Exception {
+        sink(filePart.getContentType());  // $ hasRemoteValueFlow
+        sink(filePart.getHeader("test")); // $ hasRemoteValueFlow
+        sink(filePart.getInputStream()); // $ hasRemoteValueFlow
+        sink(filePart.getHeaders("test")); // $ hasRemoteValueFlow
+        sink(filePart.getHeaderNames()); // $ hasRemoteValueFlow
+        sink(filePart.getSubmittedFileName()); // $ hasRemoteValueFlow
+        sink(filePart.getName()); // $ hasRemoteValueFlow
```

Comment:
Please sort in the same order as the models (alphabetically). Apply this below too.

Reply:
```suggestion
        sink(filePart.getContentType());  // $ hasRemoteValueFlow
        sink(filePart.getHeader("test")); // $ hasRemoteValueFlow
        sink(filePart.getHeaderNames()); // $ hasRemoteValueFlow
        sink(filePart.getHeaders("test")); // $ hasRemoteValueFlow
        sink(filePart.getInputStream()); // $ hasRemoteValueFlow
        sink(filePart.getName()); // $ hasRemoteValueFlow
        sink(filePart.getSubmittedFileName()); // $ hasRemoteValueFlow
```